### PR TITLE
Issue #166 Dropdown 

### PR DESCRIPTION
**Pull request by adamkendis**
#166

Does NOT fully close the issue. Datepicker will be pushed tomorrow, 2/11 (probably).

Dropdown adhering to Bulma style modifiers. Example:

```
    <Dropdown
      id="datefilter-dropdown"
      list={[
        { label: 'Last Week', value: 'LAST_WEEK' },
        { label: 'Last Month', value: 'LAST_MONTH' },
        { label: 'Last 6 Months', value: 'LAST_6_MONTHS' },
      ]}
      title="Select Date Range"
      handleClick={someFunc}
      style={{ left: '550px' }}
    />
```

The thought is that the value of each option will correspond with a switch case in a reducer. See props for additional Bulma modifiers.

### CODE REVIEW

File: src/components/common/Dropdown.jsx [hunk 1]
```diff
@@ -1,36 +1,146 @@
...
+  const [isOpen, updateIsOpen] = useState(!!open);
+  const [currentSelection, updateSelection] = useState(title);
+  const dropdownClassName = classNames('dropdown', {
+    'is-active': isOpen,
+    'is-hoverable': hoverable,
+    'is-right': rightAligned,
+    'is-up': dropUp,
+  });
+
+  const dropdownWidth = '350px';
```
Comment (brodly) [inline src/components/common/Dropdown.jsx]: can this be a prop?

File: src/components/common/Dropdown.jsx [hunk 2]
```diff
@@ -1,36 +1,146 @@
...
+  title,
+  handleClick,
+  style,
+  open,
+  hoverable,
+  rightAligned,
+  dropUp,
+}) => {
+  const dropdownNode = useRef();
+  const [isOpen, updateIsOpen] = useState(!!open);
```
Comment (brodly) [inline src/components/common/Dropdown.jsx]: what is the need for !! here

File: src/components/common/DropdownItem.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  label,
+  value,
+  active,
+  handleClick,
+}) => {
+  const itemClassName = classNames('dropdown-item', {
+    'is-active': active,
+  });
+
+  const itemWidth = '350px';
```
Comment (brodly) [inline src/components/common/DropdownItem.jsx]: can this be a prop too?

File: src/components/common/Dropdown.jsx (hunk 1 above)
Comment (adamkendis) [reply src/components/common/Dropdown.jsx]: Added `width` prop to Dropdown with defaultProp set to 350px.

File: src/components/common/Dropdown.jsx (hunk 2 above)
Comment (adamkendis) [reply src/components/common/Dropdown.jsx]: Thanks for catching that. Double bang operator removed. The intent was to pass `false` to useState if the open prop was not provided (undefined -> true -> false), however I'm now realizing...
1. If not passed, `open` === undefined which is already falsy, and
2. defaultProp for `open` is false

File: src/components/common/DropdownItem.jsx (hunk 3 above)
Comment (adamkendis) [reply src/components/common/DropdownItem.jsx]: Added `width` as required prop to DropdownItem. Items will receive width from Dropdown (either default Dropdown value or whatever value is passed in props).

**Pull request merged**